src/test/scala/dibl/LinkedNodeSpec.scala

```diff
@@ -0,0 +1,173 @@
+package dibl
+
+import dibl.LinkProps.simpleLink
+import org.scalatest.{ FlatSpec, Matchers }
+
+class LinkedNodeSpec extends FlatSpec with Matchers {
+
+  "Diagram.tileLinks" should "determine clockwise by over/under rules" in {
+    // from patchWidth=6&patchHeight=5&tile=5-&tileStitch=ct&&shiftColsSW=-1&shiftRowsSW=1&shiftColsSE=1&shiftRowsSE=1
+    val diagram = Diagram(
+      Seq(
+        twist("a12", 4, 4), // 0
+        twist("a11", 8, 4), // 1
+        cross("a10", 6, 6), // 2
+        twist("a11", 6, 6), // 3
+        twist("a12", 6, 6), // 4
+        cross("a10", 4, 8), // 5
+        twist("a12", 4, 8), // 6
+        cross("a10", 8, 8), // 7
+        twist("a11", 8, 8), // 8
+      ), Seq(
+        whiteStart(0, 2),
+        simpleLink(1, 2),
+        simpleLink(0, 3),
+        whiteStart(2, 3),
+        simpleLink(2, 4),
+        whiteStart(1, 4),
+        simpleLink(3, 5),
+        whiteStart(3, 6),
+        whiteStart(4, 7),
+        simpleLink(4, 8),
+      )
+    )
+    val tileLinks = diagram.tileLinks(6, 6, 6, 6)
+    tileLinks.map(_.core) shouldBe Seq(2,3,4).map(diagram.nodes)
+    tileLinks.map(_.clockwise.map(diagram.nodes.indexOf)).toArray shouldBe
+      Array(Array(0,1,4,3), Array(0,2,6,5), Array(2,1,7,8)) // node numbers
```

Comment:
Might have developed a blind spot for mistakes in the test data (line 12-31), [expected and actual](https://travis-ci.org/d-bl/GroundForge/jobs/478732911#L1310) results.

Visualizing the intended test data: purple=index, blue=x/y, red=id
![](https://raw.githubusercontent.com/d-bl/GroundForge/9fafb4108962fa8100611e3ae073188992eb7117/src/test/scala/dibl/LinkedNodeSpec.png)